### DISCUSSION TITLE
fwupd should check that SkuSiPolicyVersion and SkuSiPolicyUpdateSigners exist as an HSI test

### DISCUSSION THREAD

**Discussion by hughsie** [fwupd/fwupd#7329] [Ideas]
On a Linux system it may be possible to use a revoked Windows bootloader to bypass secure boot as the SkuSiPolicyVersion and SkuSiPolicyUpdateSigners (EFI_VARIABLE_BOOTSERVICE_ACCESS) variables have not been created by Windows (as it doesn't exist).

I think if we don't find any Windows bootloader in any of the ESPs and those UEFI keys are not found we should fail the HSI tests (HSI-2 _perhaps_) and then as a `security-fix` action offer to create them with the dummy contents as supplied by Microsoft -- e.g. see: https://support.microsoft.com/en-us/topic/kb5027455-guidance-for-blocking-vulnerable-windows-boot-managers-522bb851-0a61-44ad-aa94-ad11119c5e91

Sane?

**Comment by superm1** [comment]
Is it a boot variable or runtime variable?  I had figured it was a boot variable.  It doesn't change our ability to do things though.  We could use fwupdx64.efi to do it..

**Reply by hughsie** [reply]
BOOT, and I'm unsure if you can *create* a BOOT variable in runtime mode -- you certainly can't edit it for obvious reasons.

**Reply by superm1** [reply]
Yeah; I think it will need to be done by fwupdx64.efi so that it's before `ExitBootServices()`.  Totally makes sense as a "security-fix" in this context.

**Reply by vathpela** [reply]
> BOOT, and I'm unsure if you can _create_ a BOOT variable in runtime mode -- you certainly can't edit it for obvious reasons.

That SetVariable() call should fail:
`• Once ExitBootServices() is performed, only variables that have EFI_VARIABLE_RUNTIME_ACCESS and
EFI_VARIABLE_NON_VOLATILE set can be set with SetVariable().`

**Reply by jsetje** [reply]
I did add support for setting those variables with shim. They can be set via `mokutil --set-ssp-policy latest ; reboot`

We should make that automatic at some point, and with a little bit of care to ensure that there isn't and active bootmgr based OS on the system that's managing those itself.

**Reply by hughsie** [reply]
@jsetje I tried this just now with `shim-x64-15.8-3.x86_64` and `mokutil-0.7.1-1.fc40.x86_64` -- I set the `SSPPolicy-605dab50-e046-4300-abb6-3dd810dd8b23` to latest  (attached) in fwupd -- but after rebooting shim deleted `SSPPolicy` but didn't set `SkuSiPolicyVersion` as expected.  Any ideas?

[SSPPolicy-605dab50-e046-4300-abb6-3dd810dd8b23.txt](https://github.com/user-attachments/files/15747432/SSPPolicy-605dab50-e046-4300-abb6-3dd810dd8b23.txt)
